src/csv/lexer.cr

```diff
@@ -89,15 +89,15 @@ abstract class CSV::Lexer
       when '\0'
         raise "unclosed quote"
         break
-      when '"'
+      when @quote_char
         case next_char
-        when ','
+        when @separator
           check_last_empty_column
           break
         when '\r', '\n', '\0'
           break
-        when '"'
-          @buffer << '"'
+        when @quote_char
```

Comment:
Oh, I see what you mean. Yes, it's safe.

Do you suggest that we could make:

```crystal
if @quote_char # or when
  @quote_char ...
end
```

work if the compiler knows `@quote_char` is read-only?